### PR TITLE
Bugfix MTE-3733/MTE-3971 testPinToShortcuts() for iOS 15 and 16

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-3733)

## :bulb: Description
`testPinToShortcuts()` fails consistently on iOS 15 and 16 only because the pushpin image 📌 does not have the identifier `StandardImageIdentifiers.Small.pinBadgeFill`.

A workaround is to test for the second image under the "Example Domain" cell.

This PR also uses `localhost` as the website instead of fetching an external one.

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/PhotonActionSheetTests.swift [hunk 1]
```diff
@@ -22,15 +22,25 @@ class PhotonActionSheetTests: BaseTestCase {
...
         cell.press(forDuration: 2)
-        app.tables.cells.otherElements[StandardImageIdentifiers.Large.pinSlash].tap()
+        app.tables.cells.otherElements[StandardImageIdentifiers.Large.pinSlash].waitAndTap()
         // Check that it has been unpinned
         /* FIXME: Adding a workaround until https://github.com/mozilla-mobile/firefox-ios/issues/22323 is fixed
          * We will wait for the pinned icon on the example.com tile to disappear (max 8 seconds polling)
          */
-        waitForNoExistence(app.cells["Example Domain"].images[StandardImageIdentifiers.Small.pinBadgeFill], timeoutValue: 8)
+        if #available(iOS 17, *) {
+            mozWaitForElementToNotExist(app.cells["Example Domain"].images[StandardImageIdentifiers.Small.pinBadgeFill])
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/PhotonActionSheetTests.swift:40]: The default `TIMEOUT` for `mozWaitForElementToNoExist` should be `15`, which should be enough even on slower machines.

**Review by dragosb01 — APPROVED** [review]

**Review by mdotb-moz — APPROVED** [review]
LGTM

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/PhotonActionSheetTests.swift [hunk 2]
```diff
@@ -22,15 +22,25 @@ class PhotonActionSheetTests: BaseTestCase {
         let itemCell = app.cells[AccessibilityIdentifiers.FirefoxHomepage.TopSites.itemCell]
         let cell = itemCell.staticTexts["Example Domain"]
         mozWaitForElementToExist(cell)
+        if #available(iOS 17, *) {
+            mozWaitForElementToExist(app.cells["Example Domain"].images[StandardImageIdentifiers.Small.pinBadgeFill])
+        } else {
+            // No identifier is available for iOS 16 amd below
```
Comment (mdotb-moz) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/PhotonActionSheetTests.swift:28]: typo in the comment

**Comment by isabelrios** [comment]
I realized the tests are using real site, I added this [ticket](https://mozilla-hub.atlassian.net/browse/MTE-3971) to try to replace that, could we use this PR that is changing tests in that test suite to get it done?

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/PhotonActionSheetTests.swift [hunk 3]
```diff
@@ -22,18 +22,27 @@ class PhotonActionSheetTests: BaseTestCase {
...
-        waitForNoExistence(app.cells["Example Domain"].images[StandardImageIdentifiers.Small.pinBadgeFill], timeoutValue: 8)
+        if #available(iOS 17, *) {
+            mozWaitForElementToNotExist(app.cells["Example Domain"].images[StandardImageIdentifiers.Small.pinBadgeFill])
+        } else {
+            mozWaitForElementToNotExist(app.cells["Example Domain"].images.element(boundBy: 1))
+        }
 
-        cell.press(forDuration: 2)
-        mozWaitForElementToExist(app.tables.cells.otherElements[StandardImageIdentifiers.Large.pin])
+        mozWaitForElementToNotExist(app.cells["Example Domain"])
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/PhotonActionSheetTests.swift:45]: When I use `localhost` as the website, the site is removed completely from top sites.

**Comment by clarmso** [comment]
> @clarmso I realized the tests are using real site, I added this [ticket](https://mozilla-hub.atlassian.net/browse/MTE-3971) to try to replace that, could we use this PR that is changing tests in that test suite to get it done?

Sounds good. I found `test-example.html` from the fixtures.

**Pull request merged**